src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -140,7 +149,42 @@ private void setEnv(EnvVars env) {
     private EnvVars getEnv() {
         return this.env;
     }
-
+    
+    // To retain Backward compatibility
+    protected Object readResolve() {
+        // Assign default values to new elements.
+        this.pdfReportArtifact = new NullArtifact();
+        this.tapArtifact = new NullArtifact();
+        this.junitArtifact = new NullArtifact();
+        this.coberturaArtifact = new NullArtifact();
+        this.stmResultsArtifact = new NullArtifact();
+        this.modelCoverageArtifact = new NullArtifact();
+
+        // Assign appropriate artifact type if it was selected earlier.
+        if (pdfReportChkBx) {
+            this.pdfReportArtifact = new PdfArtifact("matlabTestArtifacts/testreport.pdf");
+        }
+        if (tapChkBx) {
+            this.tapArtifact = new TapArtifact("matlabTestArtifacts/taptestresults.tap");
+        }
+        if (junitChkBx) {
+            this.junitArtifact = new JunitArtifact("matlabTestArtifacts/junittestresults.xml");
+        }
+        if (coberturaChkBx) {
+            this.coberturaArtifact = new CoberturaArtifact("matlabTestArtifacts/cobertura.xml");
+        }
+        if (stmResultsChkBx) {
+            this.stmResultsArtifact =
+                    new StmResultsArtifact("matlabTestArtifacts/simulinktestresults.mldatx");
+        }
+        if (modelCoverageChkBx) {
+            this.pdfReportArtifact =
+                    new ModelCovArtifact("matlabTestArtifacts/coberturamodelcoverage.pdf");
+        }

```

Comment:
Why is this creating a `pdfReportArtifact`?

Reply:
Nice Catch :P  Updated it

Reply:
Shouldn't it have `.xml` extension as well?

Reply:
Yep :) done 1

Reply:
Also, why is this a different value than the default in jelly? One is `coberturamodelcoverage.xml` and the other is `modelcoverage.xml`. Seems like these should be the same.

https://github.com/mathworks/jenkins-matlab-plugin/blob/a8f6ce39a9582e6dc887c7f69b9dc20220a254af/src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly#L38-L42

Reply:
This change is already merged in last PR #92

Reply:
Which one is right?

Reply:
The One which is merged is correct . That why i did not include that file in this PR   	
below is correct and already merged in the qualification branch 
  <f:textbox default="matlabTestArtifacts/coberturamodelcoverage.xml"/>

Reply:
https://github.com/mathworks/jenkins-matlab-plugin/blob/6f5ac8c8a322a8b96ae1d1e1df65178e18c2ac3c/src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly#L40